DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs

```diff
@@ -50,5 +51,38 @@ public string RegisterDelegate(DelegateRegistrationModel delegateRegistrationMod
 
             return candidateNumber;
         }
+
+        public int RegisterCentreManagerAdmin(RegistrationModel registrationModel)
+        {
+            var values = new
+            {
+                forename = registrationModel.FirstName,
+                surname = registrationModel.LastName,
+                email = registrationModel.Email,
+                password = registrationModel.PasswordHash,
+                centreID = registrationModel.Centre,
+                centreAdmin = 1,
+                isCentreManager = 1,
+                approved = 1,
+                active = 1
+            };
+
+            var adminUserId = connection.QuerySingle<int>(
+                @"INSERT INTO AdminUsers (Forename, Surname, Email, Password, CentreId, CentreAdmin, IsCentreManager, Approved, Active)
+                        OUTPUT Inserted.AdminID
+                      VALUES (@forename, @surname, @email, @password, @centreId, @centreAdmin, @isCentreManager, @approved, @active)",
+                values
+            );
+            connection.Execute(
+                @"INSERT INTO NotificationUsers (NotificationId, AdminUserId)
+                SELECT N.NotificationId, @adminUserId
+                FROM Notifications N INNER JOIN NotificationRoles NR 
+                ON N.NotificationID = NR.NotificationID 
+                WHERE RoleID IN (1,2) AND AutoOptIn = 1",
+                new { adminUserId }
+            );

```

Comment:
This should go in a transaction

Reply:
Ah never mind, the calling code has the transaction. In that case, these should go in separate methods (maybe even in separate data services if there is an appropriate place to put the second query?)

Reply:
Initially I'd put them in separate data services but Stella said adding the default notification preferences should always happen when adding an admin to the database, so they can be together (like how the stored procedure for registering delegate sets default notification preferences).

Reply:
I'm not keen on the fact that it wasn't clear that there was a transaction going on here with 2 separate executions (as seen from my comments). But if @stellake thinks this is OK that's fine I guess.

Reply:
I think these should be in one method because these should always be done hand-in-hand. I do agree with Alex that the 2 separate executions without transaction wrapping can be confusing - we should be able to nest transactions as per: https://docs.microsoft.com/en-us/previous-versions/dotnet/articles/ms973865(v=msdn.10)?redirectedfrom=MSDN#working-with-systemtransactions